firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -142,22 +142,25 @@ public InstallationResponse createFirebaseInstallation(
         httpsURLConnection.addRequestProperty(X_ANDROID_IID_MIGRATION_KEY, iidToken);
       }
 
-      GZIPOutputStream gzipOutputStream =
-          new GZIPOutputStream(httpsURLConnection.getOutputStream());
+      GZIPOutputStream gzipOutputStream = null;
       try {
+        gzipOutputStream = new GZIPOutputStream(httpsURLConnection.getOutputStream());
         gzipOutputStream.write(
             buildCreateFirebaseInstallationRequestBody(fid, appId).toString().getBytes("UTF-8"));
       } catch (JSONException e) {
         throw new IllegalStateException(e);
       } finally {
-        gzipOutputStream.close();
+        if (gzipOutputStream != null) {
+          gzipOutputStream.close();
+        }
       }
 
       int httpResponseCode = httpsURLConnection.getResponseCode();
 
       if (httpResponseCode == 200) {
         return readCreateResponse(httpsURLConnection);
```

Comment:
I take it readCreateResponse() calls close itself?

Reply:
rather, .disconnect()

Reply:
I think it would be cleaner to have the reader method not disconnect, then you could do it in one place, maybe in a finally.

Reply:
Makes sense. Updated. PTAL.